imitation/model_env.py

```diff
@@ -0,0 +1,346 @@
+"""Finite-horizon discrete environments with known transition dynamics. These
+are handy when you want to perform exact maxent policy optimisation."""
+
+import abc
+
+import gym
+from gym import spaces
+import numpy as np
+
+
+class ModelBasedEnv(gym.Env, metaclass=abc.ABCMeta):
```

Comment:
Is there a good reason why we should be using `ABCMeta` here? Otherwise I'd just inherit from `abc.ABC`.